packages/grid/x-data-grid/src/models/colDef/gridColDef.ts

```diff
@@ -257,12 +253,26 @@ export interface GridActionsColDef<R extends GridValidRowModel = any, V = any, F
   getActions: (params: GridRowParams<R>) => React.ReactElement<GridActionsCellItemProps>[];
 }
 
+export interface GridSingleSelectColDef<R extends GridValidRowModel = any, V = any, F = V>
+  extends GridBaseColDef<R, V, F> {
+  /**
+   * Type allows to merge this object with a default definition [[GridColDef]].
+   * @default 'singleSelect'
+   */
+  type: 'singleSelect';
+  /**
+   * To be used in combination with `type: 'singleSelect'`. This is an array (or a function returning an array) of the possible cell values and labels.
+   */
+  valueOptions?: Array<ValueOptions> | ((params: GridValueOptionsParams<R>) => Array<ValueOptions>);
```

Comment:
It's not documented anymore in https://deploy-preview-7685--material-ui-x.netlify.app/x/api/data-grid/grid-col-def/ (same as `getActions` not being documented there).
We can address this later